### PR TITLE
Attempt at a client

### PR DESCRIPTION
**Pull request by grahamjenson**
This is a very early implementation of a client, it is not well tested but works for me currently (perhaps better with my other PRs)

I think this would be a good base to build from, so I am sharing now as a means of communicating progress. 

If merged, I think we should also include some docs and a disclaimer that this is still very beta.

Graham

### CODE REVIEW

**Comment by erkie** [comment]
I love this attempt at making a client, since it is a very requested feature. Would it be possible for you to add the documentation mentioned below?

> If merged, I think we should also include some docs and a disclaimer that this is still very beta.

**Comment by grahamjenson** [comment]
@erkie Added some example code and a warning. This is still very basic and I will improve soon.

**Comment by erkie** [comment]
> but merging might let others contribute as well :)

100% my thoughts as well. 

Closes #583
Closes #558 
Closes #526

File: client.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+type Client struct {
+	conn      *conn
+	namespace string
+	handlers  *namespaceHandlers
+	url       string
+	opts      *engineio.Options
+}
+
+// NewServer returns a server.
+func NewClient(uri string, opts *engineio.Options) (*Client, error) {
```
Comment (sshaplygin) [inline client.go:27]: it is so awful public API for client :( please looks [this](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) article

Comment (grahamjenson) [reply client.go:27]: Please fix it, PRs welcome :)

File: client.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+	}
+
+	fmt.Println(client)
+
+	return client, nil
+}
+
+func (s *Client) Connect() error {
+	dialer := engineio.Dialer{
+		Transports: []transport.Transport{polling.Default},
```
Comment (lettered) [inline client.go:58]: This place is written dead, the incoming opts should be read.
``` 
dialer := engineio.Dialer{
    Transports: []transport.Transport{websocket.Default},
}
if len(c.opts.Transports) > 0 {
    dialer.Transports = c.opts.Transports
}
```

Comment (lettered) [reply client.go:58]: Maybe you get a better idea.

Comment (introspection3) [reply client.go:58]: please fix it sir

**Pull request merged**